### ISSUE TITLE
Make config override env names allow upper case

### ISSUE THREAD

**Issue by lenny-intel** [edgexfoundry/go-mod-bootstrap#69]
Currently the config override env names as `PascalCase` which is non-standard. Changing the names to UPPERCASE only is a breaking change. For now allow use of UPPERCASE names in addition to the current PascalCase names until release 2.0. At that time we can force only UPPERCASE.

**Comment by tonyespy** [comment]
> Currently the config override env names as PascalCase which is non-standard.

Non-standard by who's definition??? Is this causing breakage somewhere?

I've definitely raised this issue in the past with respect to the fact that there are couple of lowercase environment variables that we currently support. I think I just gave up and didn't mention the camel case when I reviewed the config-seed ADR.

If we're going to convert everything to UPPERCASE in Hanoi, why bother with this now???
